backend/src/AppServer.hs

```diff
@@ -49,50 +46,36 @@ ratingAdjustment winner loser
     diff = abs (winner - loser)
     (smallAdjust, bigAdjust) = maybe maxThreshold snd (Map.lookupGE diff ratingThresholds)
 
-submitReportHandler :: GameReport -> AppM SubmitGameReportResponse
-submitReportHandler r = case validateReport r of
+submitReportHandler :: RawGameReport -> AppM SubmitGameReportResponse
+submitReportHandler report = case validateReport report of
   Failure errors -> throwError $ err422 {errBody = show errors}
-  Success (GameReport {..}) -> do
-    env <- ask
-    liftIO . withResource env.dbPool $ \conn -> withTransaction conn $ do
```

Comment:
It's been fun watching the progression of `conn` being shunned more and more with each refactor 😂

Reply:
Hahaha, man, screw conn. All my friends hate conn.

Nah for real though, that's sort of a neat observation. I like that that's what happening, because managing a connection to the database really feels like something that Haskell should be doing for me (via some monad), and not something I should be doing by hand. Fortunately, that's now exactly the nice behavior we have!

Reply:
I keep using "monad", "context", and "monadic context" kind of interchangeably - but I really do think "a context in which a computation happens" is the closest thing we can have to an "intuitive" understanding of such an abstract concept of a monad.

In this case, it's pretty educational to look at the type of `runDb` and see that its single parameter is of type `SqlPersistT IO a` (which is to say, an SqlPersist monad *transformer* that wraps an IO monad). The entire `do` block is the argument given to `runDb`, so the whole block happens within the `SqlPersistT` context. And what is that context? It's "a computation that has access to a database connection, where all queries on the connection happen in a single transaction". A very nice monad indeed!

Reply:
"A context in which a computation happens" is a helpful way to think about it for me! It's an awesome monad 👏